docs/content/docs/contributor-docs/code-generator-config.md

```diff
@@ -694,19 +706,83 @@ We set this `requeue_on_success_seconds` value to `60` here because the values o
+
+Use the `resources[$resource].fields[$field].is_read_only` configuration option to override whether a field should go in the `Status` struct.
+
+Here is an example from the Lambda controller's [generator.yaml](https://github.com/aws-controllers-k8s/lambda-controller/blob/2ee7a2969ee23c900a18f6265a272669b058b62e/generator.yaml#L52-L53) file that instructs the code generator to treat the `LayerStatuses` field as a read-only field (and thus should belong in the `Status` struct for the Function resource):
+
+```yaml
+resources:
+  Function:
+    fields:
+      LayerStatuses:
+        is_read_only: true
+```
+
+Typically, you will see this configuration option used for fields that have two different Go types representing the modifiable version of the field and the non-modifiable version of the field (as is the case for a Lambda Function's Layers information) or when you need to create a custom field.
 
 ### Marking a field as required
 
-**TODO**
+If an AWS API model file marks a particular member field as required, `ack-generate` will usually infer that the associated custom resource field is required. Sometimes, however, you may want to override whether or not a field should be required. Use the `resources[$resource].fields[$field].is_required` configuration option to do so.
+
+Here is an example from the EC2 controller's [`generator.yaml`](https://github.com/aws-controllers-k8s/ec2-controller/blob/3b1ba705df02f9b7db1a8079ed7729af7a4f213a/generator.yaml#L206-L209) file that instructs the code generator to treat the Instance custom resource's MinCount and MaxCount fields as **not required**, even though the API model definition marks these fields as required in the Create Operation's Input shape.
+
+NOTE: The reason for this is because the EC2 controller only deals with single Instance resources, not batches of instances
+
+```yaml
+resources:
+  Instance:
+    fields:
+      MaxCount:
+        is_required: false
+      MinCount:
+        is_required: false
+```
 
 ### Controlling a field's Go type
 
-**TODO**
+Use the `resources[$resource].fields[$field].type` configuration option to override a field's Go type. You will typically use this configuration option for custom fields that are not inferred by `ack-generate` by looking at the AWS API model definition.
+
+An example of this is the Policies field for a Role custom resource definition in the IAM controller. The IAM controller uses some custom hook code to allow a Kubernetes user to specify one or more Policy ARNs for a Role simply by specifying `Spec.Policies`. To define this custom field as a list of string pointers, the IAM controller's [`generator.yaml`](https://github.com/aws-controllers-k8s/iam-controller/blob/b6a62ca48c1aea7ab78e62fb3ad6845335c1f1c2/generator.yaml#L164-L168) file uses the following:
```

Comment:
Add some language here explaining that you have chosen to "flatten" the business logic around associating Policies with a Role here by implementing a custom field, and that this field is not actually present upstream in the API.